op-acceptance-tests: interop sequencing window expiry test

protolambda opened the pull request:
**Description**

This PR tests the sequencing window expiry in an interop network.

It includes some devstack option configuration improvements, that I may split out into its own PR.

The test is not passing yet; without the L2 tx and cross-safe block at the start is passes (while doing the expiry reorg as expected), but with it included (more similar to a real network) it fails.

It appears to correctly detect sequencing window expiry in the op-node, and forces empty blocks.
Those blocks then fail to consolidate with the pre-existing unsafe blocks, and thus create a reorg.

The supervisor then detects the inconsistency, but instead of following the post-reorg chain it seems to want to stick to the pre-reorg chain, and issues a reset to the op-node to sync to the conflicting block as target, instead of a common block.
Ideally the supervisor issues no reset at all; the unsafe chain should be rewound, and the local-safe chain was extended with the force-derived block; the op-node may differ with the older op-supervisor state, but the supervisor has no reason to stay on the old chain.

Possible things that are going wrong:
- Reset-tracker bug in supervisor; it should prioritize the local-safe chain, but maybe it does not properly
- It tries to target a known conflicting block, the one of the chain before the reorg, the first conflicting block, even if the reset was determined correctly, this is not right to target.

I think we should park this PR, and invest in some improvements of the `resetTracker`. It overlaps with the upgrade-activation support work. And once that is cleaned up there is a good chance that this works properly.

**Tests**

Test-only

**Additional context**

<!--
Add any other context about the problem you're solving.
-->

**Metadata**

Fix https://github.com/ethereum-optimism/optimism/issues/16000

protolambda committed:
op-acceptance-tests: interop sequencing window expiry test

ajsutton commented on op-acceptance-tests/tests/interop/seqwindow/expiry_test.go:
```diff
@@ -0,0 +1,104 @@
+	require.Equal(safe.Derived.Hash, other.Hash, "supervisor must match chain with EL")
+
+	// re-enable the batcher now that we are done with the test.
+	t.Require().NoError(sys.L2BatcherA.ActivityAPI().StartBatcher(t.Ctx()))
```
Is it worth submitting another L2 tx after the sequencer is deriving again to make sure we really do fully recover and start including new tx and getting them into batches so they become safe (rather than them being re-orged out again because the sequencer window keeps expiring)?

protolambda replied:
I think there's a rabbit-hole there with auto-enabling sequencer recovery mode. But we can do that, or make the test enable it via RPC. I'll add a TODO in the main issue. I've split out the test as separate issue.

ajsutton replied:
Ah, I don't want to create rabbit holes - it seemed like a relatively quick win on the surface.  Probably is worth considering as a future improvement but don't take it as a blocker unless you feel its necessary to give us sufficient confidence that we can handle a sequencer window expiry.

ajsutton reviewed (approved):
LGTM. I would suggest adding a skip at the point where this fails and getting it merged. That way you won't get conflicts with other DSL changes and its easy for people working to fix issues to run the test without maintaining multiple branches or stacking PRs.

protolambda commented:
Rebased, and fixed the minor conflicts with the develop DSL changes. I've added a test-skip as per review recommendation, and I've split the issue up (this now fixes the testing part, but implementation work remains tracked).
